crates/redeem/src/tests.rs

```diff
@@ -60,6 +60,7 @@ fn test_request_redeem_fails_with_amount_exceeds_user_balance() {
                 to_be_issued_tokens: 0,
                 issued_tokens: 10,
                 to_be_redeemed_tokens: 0,
+                replace_collateral: 0,
```

Comment:
Does it make sense to refactor this in this PR to have an empty vault that you insert and you'd just specify anything that the test really needs? So maybe having a default implementation on the vault might make sense and then just update the things you need.

Very loosely something like

```rust
vault = Vault {
  issue_tokens: 10,
  ...
}

Reply:
That's how we do it in the integration tests now, so yes it makes sense to refactor this. But to be honest the unit tests are in quite a bad shape, they need more cleaning up than just changing these structs. I would prefer leaving this for another PR